backend/server.js

```diff
@@ -0,0 +1,94 @@
+import express from "express";
+import cors from "cors";
+import mongoose from "mongoose";
+import nesGamesData from "./data/NES-games.json";
+
+const port = process.env.PORT || 8080;
+const app = express();
+
+app.use(cors());
+app.use(express.json());
+
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: "Service unavaliable" })
+  }
+});
+
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.Promise = Promise;
+
+// MongoDB 
+const Game = mongoose.model("Game", {
+  title: String,
+  developer: String,
+  publisher: String,
+  release_date: String
+});
+
+if (true) {
+  const resetDatabase = async () => {
+    await Game.deleteMany();
+    nesGamesData.forEach(singleGame => {
+      const newGame = new Game(singleGame);
+      newGame.save();
+    })
+  }
+  resetDatabase();
+}
+
+// Start defining your routes here
+app.get("/", (req, res) => {
+  res.json({responseMessage: "NES-games library at /games"});
+});
+
+app.get("/games/", async (req, res) => {
+  const {title, developer, publisher} = req.query;
+
+  const titleQuery = title ? title : /.*/gm;
+  const developerQuery = developer ? developer : /.*/gm;
+  const publisherQuery = publisher ? publisher : /.*/gm;
+
+  try {
+    const response = await Game.find({title: titleQuery, developer: developerQuery, publisher: publisherQuery});
+    res.status(200).json({
+      success: true,
+      nesGames: response
+    })
+  } catch (err) {
+    res.status(400).json({ 
+      success: false,
+      error: "Invalid id-request"
+    })
+  }
+});
```

Comment:
I'm not at all sure I'm right, but perhaps the error msg could be altered a bit, since the id is not directly involved here. 

Also, I'm wondering if it could be possible to add some alternative error msg, since an empty array is returned with success "true", if I for example misspell a developer (using the endpoint /games?developer=*developer*). Perhaps, something like if (response.body === 0) { sucess: false etc.